docs/gss.md

```diff
@@ -0,0 +1,305 @@
+`badgeColor` that looks something like:
+
+```html
+<select name="badgeColor">
+    <option value="#ff0000">Red</option>
+    <option value="#00ff00">Green</option>
+    <option value="#0000ff">Blue</option>
+</select>
+```
+
+## Validator
+
+Although we use a particular industry standard JSON Schema validator ([AJV](https://github.com/epoberezkin/ajv)), the
+validation functions provided in this package standardise validation errors to conform to our own independent format, as
+shown here:
+
+```json
+
+{
+    "isValid": false,
+    "errors": [
+        {
+            "dataPath": ["requiredField"],
+            "schemaPath": ["properties", "requiredField", "required"],
+            "rule": { "required": true },
+            "message": "validation-required-field-missing"
+        },
+        {
+            "dataPath": ["deep", "booleanField"],
+            "schemaPath": ["properties", "deep", "properties", "booleanField", "type"],
+            "rule": { "type": "boolean" },
+            "message": "validation-invalid-field-type"
+        }
+    ]
+}
+```
+
+The two top level elements are:
+
+* `isError`: A boolean indicating whether or not there were validation errors.
+* `errors`: An array containing individual errors (see below).
+
+An individual error contains the following:
+
+* `dataPath`: EL Path segments to the content which breaks a validation rule, relative to the object being
+  validated.  Intended for use with functions like `fluid.get`, which can accept an array of EL path segments.
+* `schemaPath`: EL Path segments to the rule which was violated by the content.
+* `rule`: The rule that was violated.
+* `message`: The i18n key for the validation error.
+
+Although you may supply i18n messages that follow whatever conventions you choose, the default validation error messages
+included with this package support the use of variables that come from the error. Continuing the above example, let's
+look at how the second failure might be used with an i18n message template.  That error looks like:
+
+```javascript
+// This would ordinarily be delivered by the validator, but is presented here in isolation for illustration purposes.
+var error = {
+    "dataPath": ["deep", "booleanField"],
+    "schemaPath": ["properties", "deep", "properties", "booleanField", "type"],
+    "rule": { "type": "boolean" },
+    "message": "validation-invalid-field-type"
+};
+
+// This is simplified for illustration purposes, this would nearly always be delivered as part of a more complex bundle.
+var messages = {
+    "validation-invalid-field-type": "The value supplied should be a(n) %rule.type."
+};
+
+var message = fluid.stringTemplate(messages[error.message], error); // The value supplied should be a(n) boolean.
+```
+
+## Schema-validated components
+
+A schema validated component is one whose structure is required to match its "schema" options, which is defined using
```

Comment:
Should refer to the dedicated schemaValidatedComponents.md docs here?

Reply:
Very good catch, seems like  broke it out and then didn't clean up.  Fixed now.